tests/publiccloud/btrfs.pm

```diff
@@ -0,0 +1,168 @@
+#
+# Copyright SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test BTRFS filesystem on SLE 16+
+#
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use utils;
+use serial_terminal qw(select_serial_terminal);
+use version_utils qw(is_sle is_sle_micro);
+
+our $root_dir = '/root';
+
+my $btrfs_binary = '/sbin/btrfs';
+my $snapper_binary = 'snapper';
+
+sub test_btrfs_scrub {
+    my ($self) = @_;
+
+    record_info("BTRFS", "Testing btrfs scrub");
+
+    assert_script_run(qq{$btrfs_binary scrub start -B -R /});
+    script_retry(qq{$btrfs_binary scrub status / | grep -E "Status:\\s+finished"}, retry => 30, delay => 20, timeout => 600);
+    assert_script_run(qq{$btrfs_binary scrub status / | grep -E "Error summary:\\s+no errors found"});
+}
+
+sub print_snapper_list_configs {
+    my ($self) = @_;
+    record_info("SNAP LISTC", script_output(qq{$snapper_binary list-configs}));
+}
+
+sub print_snapper_list {
+    my ($self) = @_;
+    record_info("SNAPPER LIST", script_output(qq{$snapper_binary list}));
+}
+
+sub reboot_instance {
+    my ($self, $instance) = @_;
+    $instance->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
+}
+
+sub test_snapper_rollback {
+    my ($self, $instance) = @_;
+
+    record_info("ROLLBACK", "Testing snapper rollback");
+
+    my $rollback_test_file = "rollback_test_file";
+    my $rollback_test_file_content = "rollback-test";
+
+    if (script_run(qq{$snapper_binary list-configs | grep -E "root\\s+| /"}) != 0) {
+        assert_script_run(qq{umount /.snapshots});
+        assert_script_run(qq{rm -r /.snapshots});
+        assert_script_run(qq{$snapper_binary create-config /});
+        $self->reboot_instance($instance);
+    }
+
+    $self->print_snapper_list_configs();
+
+    $self->print_snapper_list();
+
+    assert_script_run(qq{$snapper_binary create -d "baseline snapshot"});
+    $self->print_snapper_list();
+
+    assert_script_run(qq{echo "$rollback_test_file_content" | tee /etc/$rollback_test_file});
+    record_info("FILE", script_output(qq{cat /etc/$rollback_test_file}));
+    assert_script_run(qq{$snapper_binary create -d "snapshot after change"});
+    assert_script_run(qq{test -f /etc/$rollback_test_file});
+    $self->print_snapper_list();
+
+    assert_script_run(qq{$snapper_binary diff 2..3 | grep -- "--- /.snapshots/2/snapshot/etc/$rollback_test_file"});
+    assert_script_run(qq{$snapper_binary diff 2..3 | grep -- "+++ /.snapshots/3/snapshot/etc/$rollback_test_file"});
+    assert_script_run(qq{$snapper_binary diff 2..3 | grep -- "@@ -0,0 +1 @@"});
+    assert_script_run(qq{$snapper_binary diff 2..3 | grep -- "+$rollback_test_file_content"});
+
+    assert_script_run(qq{$snapper_binary rollback 2});
```

Comment:
This will break when we get images that already contain snapper images.

Is there a way how we can extract the index from the previous commands instead of assuming that there would not be any?